### ISSUE TITLE
About flow_vgg16

### ISSUE THREAD

**Issue by TiJoy** [bryanyzhu/two-stream-pytorch#35]
Thanks for your open source code! And I meet a problem on flow_vgg16. I train flow_vgg16 by
`python main_single_gpu.py /data/ltj/tsn -m flow -a flow_vgg16 --new_length=10 --epochs 350 --lr 0.001 --lr_steps 200 300`
but the result of last epoch on training process is just
`* Prec@1 48.744 Prec@3 69.469`
and I think I can't obtain 80% even though I run temporal_demo.py.
Can you give me some advice?
And I just change one line of your code:
`rgb_weight_mean = torch.mean(rgb_weight, dim=1, keepdim=True)`
it is in function `change_key_name` of `flow_vgg16.py`. I just add `keepdim=True` because it will squeeze on dim 1 if I don't add it.

And I have another question about
`clip_mean = [0.5, 0.5] * args.new_length
clip_std = [0.226, 0.226] * args.new_length`
it is in main_single_gpu.py. I want to know why the `clip_main` and `clip_std` need to multiply by `new_length`. Shouldn't the mean value be constant even though there are 10 samples?

**Comment by bryanyzhu** [comment]
Sorry for the late reply, I don't have any specific suggestions for training, maybe you already figure it out. And yes, `keepdim=True` needs to change due to different versions of PyTorch.

For the clip_mean problem. Multiply it with 10 just extends the list, we are not multiplying in the value. 
``[0.5,0.5] will become [0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5,0.5]``. This is a property of python list.

**Comment by TiJoy** [comment]
Yes I obtained results that look like normal. I tried several different learning-rate values and it proved that 0.01 is a good setting for the beginning. Maybe the frame VGG16 is the main reason.

And I searched `[0.5, 0.5] * args.new_length` later, maybe I need to learn python systematically for now.

Thanks for your reply anyway.

**Comment by bryanyzhu** [comment]
Glad you figured it out, sorry again. Closing the issue for now.
